catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java

```diff
@@ -274,10 +275,14 @@ private String createFilter(String pidKey) {
 
     private BundleContext getBundleContext() {
       Bundle bundle = FrameworkUtil.getBundle(AdminPollerServiceBean.class);
+      BundleContext context = null;
       if (bundle != null) {
-        return bundle.getBundleContext();
+        context = bundle.getBundleContext();
       }
-      return null;
+      if (context == null) {
+        throw new IllegalStateException("Error getting bundle context");
+      }
+      return context;
```

Comment:
✏️ Perhaps inverting some of these would make the method easier to follow? It just feels a little odd to have one negative check and then one positive check instead of following one way consistently.
```java
    private BundleContext getBundleContext() {
      Bundle bundle = FrameworkUtil.getBundle(AdminPollerServiceBean.class);
      if (bundle == null) {
        throw new IllegalStateException("Error getting bundle context");
      }
      
      BundleContext context = bundle.getBundleContext();
      if (context == null) {
        throw new IllegalStateException("Error getting bundle context");
      }
      
      return context;
    }
```

Or if you prefer a more functional approach with `Optional` perhaps:
```java
    private BundleContext getBundleContext() {
      return Optional.of(AdminPollerServiceBean.class)
              .map(FrameworkUtil::getBundle)
              .map(Bundle::getBundleContext)
              .orElseThrow(() -> new IllegalStateException("Error getting bundle context"));
    }
```

Reply:
I like the functional option. Updated here and in AdminHelper.